pkg/systemd/hwinfo.jsx

```diff
@@ -192,173 +184,164 @@ class CPUSecurityMitigationsDialog extends React.Component {
-                                <DataListCell key="primary content">
-                                    <span>
-                                        <div className='nosmt-heading'>{ _("Disable simultaneous multithreading") } (nosmt)</div>
-                                        <small className='nosmt-read-more-link'>
-                                            <a href="https://access.redhat.com/security/vulnerabilities/L1TF" target="_blank" rel="noopener noreferrer">
-                                                <ExternalLinkAltIcon /> { _("Read more...") }
-                                            </a>
-                                        </small>
-                                    </span>
-                                </DataListCell>,
-                            ]}
-                        />
-                        <DataListAction>
-                            <div id="nosmt-switch">
-                                <Switch isDisabled={this.state.rebooting}
-                                        onChange={ (_event, value) => this.setState({ nosmt: value }) }
-                                        isChecked={ this.state.nosmt } />
-                            </div>
-                        </DataListAction>
-                    </DataListItemRow>
-                </DataListItem>
-            );
+    const footer = (
+        <>
+            <Button variant='danger' isDisabled={rebooting || nosmt === undefined} onClick={saveAndReboot}>
+                { _("Save and reboot") }
+            </Button>
+            <Button variant='link' className='btn-cancel' isDisabled={rebooting} onClick={Dialogs.close}>
+                { _("Cancel") }
+            </Button>
+        </>
+    );
 
-        const footer = (
+    return (
+        <Modal isOpen id="cpu-mitigations-dialog"
+               position="top" variant="medium"
+               footer={footer}
+               onClose={Dialogs.close}
+               title={ _("CPU security toggles") }>
             <>
-                <Button variant='danger' isDisabled={this.state.rebooting || this.state.nosmt === undefined} onClick={this.saveAndReboot}>
-                    { _("Save and reboot") }
-                </Button>
-                <Button variant='link' className='btn-cancel' isDisabled={this.state.rebooting} onClick={Dialogs.close}>
-                    { _("Cancel") }
-                </Button>
+                <Text className='cpu-mitigations-dialog-info' component={TextVariants.p}>
+                    { _("Software-based workarounds help prevent CPU security issues. These mitigations have the side effect of reducing performance. Change these settings at your own risk.") }
+                </Text>
+                <DataList>
+                    { rows }
+                </DataList>
+                { alert !== undefined &&
+                <Alert variant="danger"
+                    actionClose={<AlertActionCloseButton onClose={() => setAlert(undefined)} />}
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19378-20230925-080729-1e17ab19-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)